Add support for `URLENCODE` function

mitya57 opened the pull request:
## Change Summary

This adds `URLENCODE` that percent-encodes special characters, so that column value can be safely substituted into URLs.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Tested on PostgreSQL. But it should work fine on other backends, too.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nocodb/nocodb?pullRequest=6831) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nocodb/nocodb?pullRequest=6831) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nocodb/nocodb?pullRequest=6831) it.</sub>

mitya57 commented:
Tested on Postgres, and it works.
![Screenshot](https://github.com/nocodb/nocodb/assets/1381584/a709c1dc-53fa-411d-954f-1bd020ed10ed)

wingkwong reviewed (changes requested):
There's a few problems in this PR.

- You can't just implement one of the DB types. If this is inevitable, we should hide `URLENCODE` option for other DB types to avoid invalid formula error. However, ideally this shouldn't happen.
- `URL_ENCODE` requires a super user role to  run `CREATE EXTENSION` in advance before using it. In real life case, the DB user (connected in NocoDB) may not have such permission and this action shouldn't happen in NocoDB in my opinion. I think this approach is a no-go.
- You may probably just use `REPLACE` or `REGEX_REPLACE` in `knex.raw` to encode those special characters specified in RFC 3986.

mitya57 commented:
> You may probably just use `REPLACE` or `REGEX_REPLACE` in `knex.raw` to encode those special characters specified in RFC 3986.

Okay, I implemented that approach. I used the list of characters in RFC 3986 section 2.2, but I removed characters that `encodeURIComponent()` does not encode, and added percent sign and space.

Please review again.

mitya57 commented:
Here is how it looks now:
![Screenshot](https://github.com/nocodb/nocodb/assets/1381584/680acb31-14ee-40dd-876e-c0aa0b6c3717)

mitya57 commented:
Rebased on top of the latest `develop` branch to fix the conflicts.
@wingkwong, @pranavxc Please review again.

mitya57 commented:
Fixed a stupid copy-paste error in tests, now they should pass.

mitya57 commented:
Improved the documentation.

mitya57 commented:
Ping please? Just rebased again.

wingkwong commented:
Re-running the tests.
